tests/load/locustfiles/locustfile.py

```diff
@@ -50,211 +72,282 @@ def _(environment, **kwargs):
 
     def on_stop(self) -> Any:
+        """Called when a User stops running."""
         if self.ws:
-            self.disconnect()
-
-    def _time_event(self, name: str) -> TimeEvent:
-        return TimeEvent(self, name)
+            self.ws.close()
+        if self.ws_greenlet:
+            gevent.kill(self.ws_greenlet)
 
-    def connect(self) -> None:
-        self.ws = create_connection(
-            self.environment.parsed_options.websocket_url,
-            header={"Origin": "http://localhost:1337"},
-            timeout=5,  # timeout defaults to None
-        )
+    def on_ws_open(self, ws: WebSocketApp) -> None:
+        """Called when opening a WebSocket.
 
-    def disconnect(self) -> None:
-        if self.ws:
-            self.ws.close()
-        self.ws = None
+        Args:
+            ws: WebSocket class object
+        """
+        self.send_hello(ws)
 
-    def hello(self) -> None:
-        """Send a 'hello' message to Autopush.
+    def on_ws_message(self, ws: WebSocketApp, data: str) -> None:
+        """Called when received data from a WebSocket.
 
-        Connections must say hello after connecting to the server, otherwise the connection is
-        quickly dropped.
+        Args:
+            ws: WebSocket class object
+            data: utf-8 data received from the server
+        """
+        message: Message = self.recv(data)
+        if isinstance(message, HelloMessage):
+            self.uaid = message.uaid
+            # In order to assure notifications are sent on startup, users are
+            # required to be subscribed to a channel
+            if not self.channels:
+                self.send_register(ws)
+            self.ws = ws  # WebSocket is available for task use
+        elif isinstance(message, NotificationMessage):
+            self.send_ack(ws, message.channelID, message.version)
+        elif isinstance(message, RegisterMessage):
+            self.channels[message.channelID] = message.pushEndpoint
+
+    def on_ws_error(self, ws: WebSocketApp, error: Exception) -> None:
+        """Called when there is a WebSocketApp error or if an exception is raised in a WebSocket
+        callback function.
 
-        Raises:
-            AssertionError: If the hello message response is empty or has an invalid status
-            ValidationError: If the hello message schema is not as expected
+        Args:
+            ws: WebSocket class object
+            error: Exception object
         """
-        body: str = json.dumps(
-            dict(
-                messageType="hello",
-                use_webpush=True,
-                uaid=self.uaid,
-                channelIDs=list(self.channels.keys()),
-            )
+        logger.error(str(error))
+
+        # WebSocket closures are expected
+        if isinstance(error, WebSocketConnectionClosedException):
+            self.ws = None  # WebSocket is not available for task use
+            self.connect()  # Reconnect
```

Comment:
I'm sure you've tested this reconnecting on WS conns closing case. However it's unclear to me how recreating the `WebSocketApp`/`run_forever` loop works with the previous instance of it spawned.

Apparently `WebSocketApp` supports reconnecting on its own but this lead me down a rabbit hole of its notion of a "dispatcher" which I'd rather not deal with. I guess this current method of reconnecting works ok? :shipit:

Reply:
Summary from Slack discussion:
It seems that we need to set the `reconnect` Arg to a value higher than 0, in order for the `run_forever()` to apply.
It doesn't seem like we need a dispatcher for our use case.